src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,112 @@
+package duke;
+
+import duke.exception.DukeDatabaseException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+/**
+ * Represents the storage that loads data from and saves data to the database file.
+ */
+public class Storage {
+    private String filePath;
+
+    /**
+     * Constructor for Storage.
+     *
+     * @param filePath the path to the txt database file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+
+    /**
+     * Load data from the txt database file.
+     *
+     * @return an array of tasks as represented by the database
+     * @throws DukeDatabaseException if database could not be created due to unknown error
+     */
+    public Task[] loadData() throws DukeDatabaseException {
+        File db = new File(this.filePath);
+        ArrayList<Task> taskList = new ArrayList<>();
+        try {
+            Scanner fileReader = new Scanner(db);
+            while (fileReader.hasNextLine()) {
+                taskList.add(readEntry(fileReader.nextLine()));
+            }
+            fileReader.close();
+        } catch (FileNotFoundException e) {
+            this.createDatabase();
+        }
```

Comment:
Hmm, is using try-catch for logic the best way to go about this? We could use File.exists() for this instead!

Reply:
Yep I agree with the suggestion! `File::exists` is really a more elegant way to handle missing file here.